src/torrent.cpp

```diff
@@ -4386,8 +4386,15 @@ namespace libtorrent {
 		// the torrent object from there
 		if (m_storage)
 		{
-			m_ses.disk_thread().async_stop_torrent(m_storage
-				, std::bind(&torrent::on_torrent_aborted, shared_from_this()));
+			try {
+				m_ses.disk_thread().async_stop_torrent(m_storage
+					, std::bind(&torrent::on_torrent_aborted, shared_from_this()));
+			} catch (std::exception const&)
+			{
+				m_storage.reset();
+				if (alerts().should_post<cache_flushed_alert>())
```

Comment:
other than the missing `\n`s in `try` and `catch`, I wonder why `cache_flushed_alert` is the right alert here

Reply:
yeah, good question. perhaps there's a better approach here. I just did the same thing as the "else" block, but failing to flush the cache is kind of serious